2023-09-18 17:48:30.706689: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 17:48:30.706888: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 17:48:38,509 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 17:48:38,591 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 17:48:38,591 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 17:48:38,591 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 17:48:38,632 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 17:48:38,633 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:38,649 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.4633333333333333}
INFO flwr 2023-09-18 17:48:38,649 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 17:48:38,910 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:38,934 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 17:48:38,936 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:38,945 | server.py:125 | fit progress: (1, 0.42308186520491603, {'accuracy': 0.81}, 0.29546320000000037)
DEBUG flwr 2023-09-18 17:48:38,945 | server.py:173 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:38,969 | server.py:187 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2023-09-18 17:48:38,970 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 17:48:38,970 | server.py:223 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:38,993 | server.py:237 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:39,004 | server.py:125 | fit progress: (2, 0.41440839565328813, {'accuracy': 0.8166666666666667}, 0.35500299999999996)
DEBUG flwr 2023-09-18 17:48:39,005 | server.py:173 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:39,023 | server.py:187 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:48:39,024 | server.py:223 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:39,047 | server.py:237 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:39,057 | server.py:125 | fit progress: (3, 0.4142287899499094, {'accuracy': 0.8166666666666667}, 0.40834480000000006)
DEBUG flwr 2023-09-18 17:48:39,057 | server.py:173 | evaluate_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:39,075 | server.py:187 | evaluate_round 3 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:48:39,076 | server.py:223 | fit_round 4: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:39,097 | server.py:237 | fit_round 4 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:39,108 | server.py:125 | fit progress: (4, 0.41422936372503555, {'accuracy': 0.8166666666666667}, 0.45886089999999946)
DEBUG flwr 2023-09-18 17:48:39,109 | server.py:173 | evaluate_round 4: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:39,126 | server.py:187 | evaluate_round 4 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:48:39,126 | server.py:223 | fit_round 5: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:39,146 | server.py:237 | fit_round 5 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:39,158 | server.py:125 | fit progress: (5, 0.41422936631355944, {'accuracy': 0.8166666666666667}, 0.5083675999999997)
DEBUG flwr 2023-09-18 17:48:39,158 | server.py:173 | evaluate_round 5: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:39,176 | server.py:187 | evaluate_round 5 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:48:39,176 | server.py:223 | fit_round 6: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:48:39,198 | server.py:237 | fit_round 6 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:39,210 | server.py:125 | fit progress: (6, 0.4142293663146087, {'accuracy': 0.8166666666666667}, 0.5609021999999992)
DEBUG flwr 2023-09-18 17:48:39,210 | server.py:173 | evaluate_round 6: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:48:39,233 | server.py:187 | evaluate_round 6 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:48:39,234 | server.py:223 | fit_round 7: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:48:39,262 | server.py:237 | fit_round 7 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:39,276 | server.py:125 | fit progress: (7, 0.4167441452745484, {'accuracy': 0.8}, 0.6270524999999996)
DEBUG flwr 2023-09-18 17:48:39,277 | server.py:173 | evaluate_round 7: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:48:39,298 | server.py:187 | evaluate_round 7 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:48:39,298 | server.py:223 | fit_round 8: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:48:39,328 | server.py:237 | fit_round 8 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:39,340 | server.py:125 | fit progress: (8, 0.41673758768933084, {'accuracy': 0.8}, 0.6905156999999988)
DEBUG flwr 2023-09-18 17:48:39,340 | server.py:173 | evaluate_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:48:39,368 | server.py:187 | evaluate_round 8 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:48:39,369 | server.py:223 | fit_round 9: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:48:39,411 | server.py:237 | fit_round 9 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:39,423 | server.py:125 | fit progress: (9, 0.41905470703297926, {'accuracy': 0.7966666666666666}, 0.7745502999999996)
DEBUG flwr 2023-09-18 17:48:39,424 | server.py:173 | evaluate_round 9: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:48:39,449 | server.py:187 | evaluate_round 9 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:48:39,450 | server.py:223 | fit_round 10: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:48:39,484 | server.py:237 | fit_round 10 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:48:39,497 | server.py:125 | fit progress: (10, 0.41905554511791093, {'accuracy': 0.7966666666666666}, 0.8477356)
DEBUG flwr 2023-09-18 17:48:39,497 | server.py:173 | evaluate_round 10: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:48:39,540 | server.py:187 | evaluate_round 10 received 5 results and 0 failures
INFO flwr 2023-09-18 17:48:39,540 | server.py:147 | FL finished in 0.8912165999999999
INFO flwr 2023-09-18 17:48:39,541 | app.py:218 | app_fit: losses_distributed [(1, 0.4286346659064293), (2, 0.4275743868201971), (3, 0.427567221224308), (4, 0.42756721563637257), (5, 0.42756721563637257), (6, 0.42906890703099115), (7, 0.4270175727350371), (8, 0.4218974170230684), (9, 0.4213106830914815), (10, 0.4213106433550517)]
INFO flwr 2023-09-18 17:48:39,541 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 17:48:39,541 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 17:48:39,541 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.42308186520491603), (2, 0.41440839565328813), (3, 0.4142287899499094), (4, 0.41422936372503555), (5, 0.41422936631355944), (6, 0.4142293663146087), (7, 0.4167441452745484), (8, 0.41673758768933084), (9, 0.41905470703297926), (10, 0.41905554511791093)]
INFO flwr 2023-09-18 17:48:39,541 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.4633333333333333), (1, 0.81), (2, 0.8166666666666667), (3, 0.8166666666666667), (4, 0.8166666666666667), (5, 0.8166666666666667), (6, 0.8166666666666667), (7, 0.8), (8, 0.8), (9, 0.7966666666666666), (10, 0.7966666666666666)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.73857813,  0.87598178,  0.75041152,  0.2783385 ,  0.2431534 ,
         0.20822857, -0.05608122, -0.12477239]]), array([-0.01353779])]
[array([[-1.73857813,  0.87598178,  0.75041152,  0.2783385 ,  0.2431534 ,
         0.20822857, -0.05608122, -0.12477239]]), array([-0.01353779])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.78331603,  0.83972291,  0.75802265,  0.24722506,  0.23657997,
         0.22524333, -0.04581329, -0.03509396]]), array([-0.04714771])]
[array([[-1.78331603,  0.83972291,  0.75802265,  0.24722506,  0.23657997,
         0.22524333, -0.04581329, -0.03509396]]), array([-0.04714771])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.78301102,  0.84043042,  0.75482329,  0.24561471,  0.23586007,
         0.22654422, -0.04425068, -0.03228206]]), array([-0.04826143])]
[array([[-1.78301102,  0.84043042,  0.75482329,  0.24561471,  0.23586007,
         0.22654422, -0.04425068, -0.03228206]]), array([-0.04826143])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.78300749,  0.84042655,  0.75478735,  0.24563225,  0.23584673,
         0.22655056, -0.04423936, -0.03228957]]), array([-0.04823657])]
[array([[-1.78300749,  0.84042655,  0.75478735,  0.24563225,  0.23584673,
         0.22655056, -0.04423936, -0.03228957]]), array([-0.04823657])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.78300746,  0.84042662,  0.75478723,  0.24563231,  0.23584669,
         0.22655057, -0.04423938, -0.03228957]]), array([-0.04823657])]
[array([[-1.78300746,  0.84042662,  0.75478723,  0.24563231,  0.23584669,
         0.22655057, -0.04423938, -0.03228957]]), array([-0.04823657])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.78300746,  0.84042662,  0.75478723,  0.24563231,  0.23584669,
         0.22655057, -0.04423938, -0.03228957]]), array([-0.04823657])]
[array([[-1.78300746,  0.84042662,  0.75478723,  0.24563231,  0.23584669,
         0.22655057, -0.04423938, -0.03228957]]), array([-0.04823657])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.86233733,  0.91504794,  0.7020193 ,  0.34006186,  0.19763699,
         0.13796051,  0.00442519,  0.01370968]]), array([-0.03167604])]
[array([[-1.86233733,  0.91504794,  0.7020193 ,  0.34006186,  0.19763699,
         0.13796051,  0.00442519,  0.01370968]]), array([-0.03167604])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.86250088,  0.91493324,  0.70203449,  0.33999623,  0.1977284 ,
         0.13792433,  0.00438755,  0.01368862]]), array([-0.03129644])]
[array([[-1.86250088,  0.91493324,  0.70203449,  0.33999623,  0.1977284 ,
         0.13792433,  0.00438755,  0.01368862]]), array([-0.03129644])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.90031712,  0.96900869,  0.70919582,  0.378721  ,  0.20792372,
         0.15547862, -0.01620562,  0.0306515 ]]), array([-0.04981263])]
[array([[-1.90031712,  0.96900869,  0.70919582,  0.378721  ,  0.20792372,
         0.15547862, -0.01620562,  0.0306515 ]]), array([-0.04981263])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.90025417,  0.96917922,  0.70943367,  0.37889051,  0.20771917,
         0.15558831, -0.01632976,  0.0309412 ]]), array([-0.05032666])]
[array([[-1.90025417,  0.96917922,  0.70943367,  0.37889051,  0.20771917,
         0.15558831, -0.01632976,  0.0309412 ]]), array([-0.05032666])]
>>>>>>>>>>>>>run time:
0.515625s
